### ISSUE TITLE
CTS scripting issue :it is double clicking the commands

### ISSUE THREAD

**Issue by ivanherrera402** [Comcast/cats#9]
From this point on, every click is performed twice.  So if I issue the command to go to channel 3 it goes to channel 33.

See email sent

Thanks,
Per
John Diaz

**Comment by cfrede001** [comment]
There are typically 2 issues that can cause this behavior:
1) IR cross talk or bounce, which can usually be eliminated by placing black tape over emitters on STBs.
2) While recording IR signals a repeat value is being included in the IR signal definition.  Below is a sample IRPacket.  Notice in the XML there is a <NoRepeats>1</NoRepeats>.  This value should typically be set to 0 or 1 and sometimes the initial recordings will indicate a larger number.
<IRPacket xsi:type="ModulatedSignal">
<Name>0</Name>
<UID>rTbrl1oFu06FrWSePPfp0g==</UID>
<PauseRepeatMode>ConstantGap</PauseRepeatMode>
<RepeatPause>0</RepeatPause>
<ModulationFreq>38000</ModulationFreq>
<Lengths>
<double>0.209</double>
<double>0.758415</double>
<double>0.895125</double>
<double>1.031835</double>
<double>1.1685450000000002</double>
<double>1.3052549999999998</double>
<double>1.441965</double>
<double>1.578675</double>
<double>1.715385</double>
<double>1.852095</double>
<double>1.988805</double>
<double>2.125515</double>
<double>2.262225</double>
<double>2.398935</double>
<double>2.535645</double>
<double>2.672355</double>
<double>2.809065</double>
<double>13</double>
</Lengths>
<SigData>
AAIACAABABAABQAFAAQADwARAAIAEAABAAEAAQABAAEAAQB/AAIACAABABAABQAFAAQADwARAAIABwAKAAEAAQABAAEAAQB/
</SigData>
<NoRepeats>1</NoRepeats>
<IntraSigPause>80</IntraSigPause>
<ToggleData/>
</IRPacket>

**Comment by ivanherrera402** [comment]
John,

FYI

Ivan

**Comment by ivanherrera402** [comment]
John,

  FYI. If you need more questions answered or if this wasn’t helpful, please follow up with Comcast. I am currently busy on another project.

Thanks,

Ivan

**Comment by ivanherrera402** [comment]
All:
I created a new issue with more info and questions:

The Remote Control use is the one that was installed by TATA/OCATS during the original installation.
There are not error when you use the virtual remote for two STBs.  Only after you execute the script.

There are only one RedRat hub process before and after executing the script.
See at the end the grep for RedRat ouput, before stopping and starting the RedRat hub.

Response for previous issue request:
There are typically 2 issues that can cause this behavior:
1) IR cross talk or bounce, which can usually be eliminated by placing black tape over emitters on STBs.
*******************
There is tape over the IR emitter.  Plus the two STBs are not next to eachother.  One is in one sede and the other is in the other side.
*********************
2) While recording IR signals a repeat value is being included in the IR signal definition. Below is a sample IRPacket. Notice in the XML there is a 1. This value should typically be set to 0 or 1 and sometimes the initial recordings will indicate a larger number.

0
rTbrl1oFu06FrWSePPfp0g==
ConstantGap
0
38000

0.209
0.758415
0.895125
1.031835
1.1685450000000002
1.3052549999999998
1.441965
1.578675
1.715385
1.852095
1.988805
2.125515
2.262225
2.398935
2.535645
2.672355
2.809065
13
********************
ANSWER:  The Remote control is the original installed installed Comcast Remote.  Does it contain the mentioned 1 or 0 or higher?
********************

Steps to recreate:

Recording the script:
After recording simple instructions see screen shot.


Actual Recorded Script

press CHUP, 10000
press CHDN, 5000
press CHDN, 9000
press CHUP, 4000
press INFO, GUIDE, EXIT
sleep 4000
press 0
press 3
sleep 5000
press LAST

Executing using same STB:
The CATSVision performed the task OK.

Executing after dding a second STB:
Adding a second STB and enabled it to work with the virtual remote.  After executing the script for both, it is double clicking the commands.
See screen capture:


From this point on, every click is performed twice.  So if I issue the command to go to channel 3 it goes to channel 33.
See Screen capture:

Questions:
1.            By executing the script for more than one STB, does the RetRat hub starts a new session?
2.            What can be done to prevent the double clicking?  Only by stopping and Starting the RedRat hub, this issue goes away.
3.            Is CATSVision not designed to execute a script for multiple STBs.
4.            If this is the case for question #3, why after some re-runs of the script with only one STB, the problem also appears?
5.            There was only one instance of RedRat.  Why is double clicking?
CentOS grep command result before and after the double clicking:
[root@localhost ~]# ps -ef |grep RedRat
root      5773  5726  0 08:30 pts/1    00:00:00 grep --color=auto RedRat
root     16796     1  0 Jul31 ?        00:07:19 mono --server /opt/ocats/software/RedratHub/RedRatHub_V4.12/RedRatHubCmd.exe /opt/ocats/cats/ocats-release/sampleConfigFiles/REDRAT_KEYMANAGER.xml --conns=400 –noscan

[root@localhost ~]# service redrathub stop
Stopping redrathub                                /etc/init.d/redrathub: line 87: cd: HOME not set
Ok
[root@localhost ~]# service redrathub start
Starting redrathub...                             Ok
[root@localhost ~]# ps -ef |grep RedRat
root      5812     1  5 08:31 pts/1    00:00:00 mono --server /opt/ocats/software/RedratHub/RedRatHub_V4.12/RedRatHubCmd.exe /opt/ocats/cats/ocats-release/sampleConfigFiles/REDRAT_KEYMANAGER.xml --conns=400 --noscan
root      5938  5726  0 08:31 pts/1    00:00:00 grep --color=auto RedRat
[root@localhost ~]#


Thanks,
John

From: Herrera, Ivan
Sent: Tuesday, August 04, 2015 9:35 AM
To: Comcast/cats; Comcast/cats
Cc: Diaz, John; ANKUSH TRAKRU; Frederick, Chad; SAJAY JANARDHANA KURUP; SREEJITH SIVASANKAR
Subject: RE: [cats] CTS scripting issue :it is double clicking the commands (#9)
Importance: Low

John,

  FYI. If you need more questions answered or if this wasn’t helpful, please follow up with Comcast. I am currently busy on another project.

Thanks,

Ivan
